### PR TITLE
Various small fixes

### LINKED ISSUE #50: Certain fonts crash get_features()
**Python version**
3.10.9

**Package version**
0.14.4

**Current behavior (bug description)**

Calling `get_features()` on [this](https://github.com/fabiocaccamo/python-fontbro/files/10505700/LeagueGothic-Regular.zip) font crashes fontbro.
Stacktrace:
```
Traceback (most recent call last):
  File "C:\Users\User\Desktop\fonts\main.py", line 24, in <module>
    font.get_features()
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\fontbro\font.py", line 319, in get_features
    features_tags = self.get_features_tags()
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\fontbro\font.py", line 338, in get_features_tags
    feature_record = table.FeatureList.FeatureRecord or []
AttributeError: 'NoneType' object has no attribute 'FeatureRecord'
```

Looks like `FeatureList` can be None. This only affects the GSUB and not the GPOS table.

### LINKED ISSUE #51: get_characters() unicode_name can be undefined
**Python version**
3.10.9

**Package version**
0.14.4

**Current behavior (bug description)**
In `get_characters()`, `unicode_name` can be undefined causing a crash. Here, an example font
[ANASTCN.zip](https://github.com/fabiocaccamo/python-fontbro/files/10514759/ANASTCN.zip) causing the crash on calling `get_characters_count()`.

Stacktrace:
```
Traceback (most recent call last):
  File "C:\Users\User\Desktop\fonts\main.py", line 30, in <module>
    main()
  File "C:\Users\User\Desktop\fonts\main.py", line 23, in main
    analyze_font("ANASTCN.ttf")
  File "C:\Users\User\Desktop\fonts\analyze_font.py", line 82, in analyze_font
    result["character_count"] = font.get_characters_count()
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\fontbro\font.py", line 311, in get_characters_count
    return len(list(self.get_characters(ignore_blank=ignore_blank)))
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\fontbro\font.py", line 295, in get_characters      
    "unicode_name": unicode_name,
UnboundLocalError: local variable 'unicode_name' referenced before assignment
```
The characters also seem to be incorrectly mapped, but I think that is an issue with the font, as fontforge shows the same incorrect mapping (though it is fine on regular use in Windows).

### LINKED ISSUE #52: Crash when glyph outside Unicode range
**Python version**
3.10.9

**Package version**
0.14.4

**Current behavior (bug description)**
Fonts my contain glyphs outside the defined Unicode range. For example, [Bobbiefreebie.ttf](https://github.com/fabiocaccamo/python-fontbro/files/10514999/Bobbiefreebie.zip) contains the ligature glyphs outside the defined range, causing `chr()` to throw an exception when calling `get_characters_count()`:
```
Traceback (most recent call last):
  File "C:\Users\User\Desktop\fonts\main.py", line 31, in <module>
    main()
  File "C:\Users\User\Desktop\fonts\main.py", line 23, in main
    result = analyze_font("Bobbiefreebie.ttf")
  File "C:\Users\User\Desktop\fonts\analyze_font.py", line 95, in analyze_font
    result["character_count"] = font.get_characters_count()
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\fontbro\font.py", line 310, in get_characters_count
    return len(list(self.get_characters(ignore_blank=ignore_blank)))
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\fontbro\font.py", line 272, in get_characters
    char = chr(code)
ValueError: chr() arg not in range(0x110000)
```

### PR DESCRIPTION
**Pull request by seproDev**
**Describe your changes**
Various small changes and fixes, the commit messages should be pretty descriptive.

**Related issue**
Closes #50, Closes #51, Closes #52 
I added the option to pass through all kwargs instead of just lazy. This partially addresses #49 through the `fontNumber` option. I think the issue should probably stay open though, since I feel like a nicer interface to interact with collections is still needed.

**Checklist before requesting a review**
- [x] I have performed a self-review of my code.
- [x] I have added thorough tests for the proposed changes.
- [ ] I have run the tests and there are no errors. (`test_get_fingerprint_diff_between_variable_instances` and `test_str` fail, ~~but they also fail on main for me~~. Only `test_str` fails in main for me.)

### CODE REVIEW

**Comment by seproDev** [comment]
Okay, seems like 1494afdb453275ae8d6a128c6729b36dfe724b2c causes the test failure. Though I am not entirely sure why. I am pretty certain the commit is correct. Should I just drop it for now?

File: fontbro/font.py [hunk 1]
```diff
@@ -271,17 +271,17 @@ def get_characters(self, ignore_blank=False):
         glyfs = font.get("glyf")
         for code, char_name in cmap.items():
             code_hex = f"{code:04X}"
-            char = chr(code)
+            if code in range(0x110000):
```
Comment (fabiocaccamo) [inline fontbro/font.py:274]: a `code_range` variable could be created outside the loop to avoid range object creation at every loop iteration.

**Review by fabiocaccamo — CHANGES_REQUESTED** [review]
Thank you very much for this great PR, just that range object optimization and I merge it.

File: fontbro/font.py (hunk 1 above)
Comment (seproDev) [reply fontbro/font.py]: I changed it to `if 0 <= code < 0x110000:` which should be cleaner.

**Pull request merged**